### PR TITLE
Engigram MVP1

### PR DESCRIPTION
**Pull request by RBGeomaticsRob**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,28 +1,70 @@
...
+bin/rake db:migrate RAILS_ENV=test
+```
+You will also be required to insert the following file on your local copy at `/config/initializers/dev_environment.rb` with an AWS S3 bucket and account details.
+```ruby
+unless Rails.env.production?
+  ENV['S3_BUCKET_NAME'] = 'xyz
+  ENV['AWS_ACCESS_KEY_ID'] = 'xyz'
+  ENV['AWS_SECRET_ACCESS_KEY'] = 'xyz'
+end
+```
```
Comment (danldb) [inline README.md]: Nice readme!

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+class CommentsController < ApplicationController
+  def create
+    post = Post.find(params[:post_id])
+    post.comments.create(feedback: params[:feedback])
+    redirect_to '/projects'
+  end
+
+  private
```
Comment (danldb) [inline app/controllers/comments_controller.rb]: Nice, skinny controller

File: app/controllers/comments_controller.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+  def review_params
+    params.require(:comment).permit(:feedback)
+  end
+
```
Comment (danldb) [inline app/controllers/comments_controller.rb]: You could put this in the CommentsHelper module

File: app/controllers/posts_controller.rb [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  def new
+    @project = Project.find(params[:project_id])
+    @post = Post.new
+  end
+
+  private
+
+  def review_params
+    params.require(:post).permit(:description, :image)
+  end
```
Comment (danldb) [inline app/controllers/posts_controller.rb]: Again, could be in a helper

File: app/controllers/projects_controller.rb [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    @project = Project.find(params[:id])
+  end
+
+  private
+
+  def project_params
+    params.require(:project).permit(:name, :company)
+  end
+
+end
```
Comment (danldb) [inline app/controllers/projects_controller.rb]: Your controllers look great - no logic, super skinny.

File: app/models/comment.rb [hunk 6]
```diff
@@ -0,0 +1,3 @@
+class Comment < ActiveRecord::Base
+  belongs_to :post
+end
```
Comment (danldb) [inline app/models/comment.rb]: Should a post also belong to a user?

**Comment by danldb** [comment]
:checkered_flag: 

Well done Rob, this is a strong submission. 

Your Travis CI is failing - the reason is that you haven't set a secret key base for Devise, so none of your devise tests are passing remotely. I normally set the secret key base in test and production and only call the environment variable in production and staging.

If you have some extra time at the end of the course, you may want to add AWS storage for the images.

Good work :)

**Comment by RBGeomaticsRob** [comment]
Thanks for taking time to review Dan, I had already implemented AWS storage and integrated with the post model is this the correct place to have done it?

**Pull request closed without merging**